_adr/90/index.adoc

```diff
@@ -0,0 +1,135 @@
+For example, if the kafka id is `12345` then Kafka boostrap host will have the form of `foobar-sqfsazk.trial.kafka.bf2.dev`.
+
+3. No migration to be performed for Kafkas.
+The proposed solution will be applied to only new Kafkas.  
+Existing Kafkas will continue to share the wildcard certificate on the `*.kafka.bf2.dev` domain.  
+The shared certificate for the already existing Kafkas will now be automatically renewed by the fleet manager.
+
+4. Certificate renewals will also be handled automatically by the fleet manager using the https://github.com/caddyserver/certmagic[certmagic library]. 
+Renewals will be blast radius aware to renew certificates in small runs on per Kafka basis.
+The intention is to reduce the risk of compromising all our certificates in case there is an outage or an issue with the renewal system in Letsencrypt. 
+It also removes the chances of bombarding Letsencrypt with bulk requests or rollout of all the Kafkas in one big batch after renewals. 
+
+5. Revocation will be handled per Kafka instance.
+It will follow a GitOps approach where SRE will request to revoke a certificate for a Kafka.
+Once the compromised certificate has been revoked, a new certificate for that Kafka will be generated.
+For already existing Kafkas, they'll continue to share the newly generated certificate: see proposal number (3). 
+
+6. When a standard Kafka is removed from the fleet manager, its corresponding certificate will also be revoked. 
+This is with an excpetion for all Kafkas that uses a shared certificate i.e Kafkas that existed before the solution is in place or all new developer Kafkas that uses a shared wildcard `*.trial.kafka.dev` certificate.
+
+7. There will be separate Letsencrypt's account per kas-fleet-manager environment. 
+For each of these acccounts, a request to overrides the various https://Letsencrypt.org/docs/rate-limits[Letsencrypt limits] will be done by following https://Letsencrypt.org/docs/rate-limits/#a-id-overrides-a-overrides[Letsencrypt limits overrides].
+There are two main limits that we should be aware of and are of interests: (1) Certificates per Registered Domain which has a default of 50 certificates per week (2) You can create a maximum of 300 New Orders per account per 3 hours.
+Each of these two limits needs to be bumped to a bigger value after capacity planning for certificate generation has been done.
+
+NOTE: There is an existing logic in the data plane that ensures that all sensitive parts in the ManagedKafka CR are not written in plain text. 
+Instead a secret is created and only references to the secrets will be added before the ManagedKafka is created in the data plane.
+
+### Threat model
+// Provide a link to the relevant threat model. 
+// You must either update an existing threat model(s) to cover the changes made by this ADR, or add a new threat model.
+
+Once the certificate are in the data plane in form of a Kubernetes secret, a priviledged users can still see them. 
```

Comment:
Is the access to these `Secret`s already controlled using RBAC? If so, let's say so explicitly.

Reply:
Yes. That's already the case for MK. 
It is called out in the line below 
```
For non enterprise data plane clusters, this is mitigated by only allowing SREs to have the priviledge / permissions to see these secrets.
```